### DISCUSSION TITLE
nerve density

### DISCUSSION THREAD

**Discussion by ayafa** [neuropoly/axondeepseg#679] [Help needed]
I am a potential new user. I am curious if you can detect nerve density and measuring individual nerve size using this software or is it mainly for detecting myelin sheaths? Thanks!

**Comment by hermancollin** [comment]
Hi @ayafa thank you for reaching out!
What do you mean exactly by nerve density? Also, IIUC, you are looking for an automated method to measure nerve area. It would be possible to train a model to segment the nerve fibers and use the segmentation to compute various metrics, although it might require some adjustments to the software. If you provide more details in regards to what you are trying to achieve, we could help you further. Keep in mind most of our dev team have a CS background, so we don't have an extensive knowledge of medical terms/methods.

**Reply by ayafa** [reply]
Hello! Thanks for your response! By nerve density, I mean does the software have the ability counts how many nerves are in a particular area? And by adjusting the software, do you mean on your end right? Like add a feature to it? Thanks!

**Reply by hermancollin** [reply]
If you provide an example image, it would be easier to assess if axondeepseg is the right tool for you.

The way ADS works is that it uses a pre-trained model to produce a segmentation. We then use this segmentation to extract various metrics. We train such a model with images acquired with a specific microscopy modality (e.g. Bright-Field Microscopy) such that the user can choose the model that fits best his data. So these models are pretty specialized, but we cover most of what researchers encounter in terms of axonmyelin segmentation and morphometrics extraction. 

If you want to segment nerves, I would recommend you use [ivadomed](https://ivadomed.org/index.html) to train your own 2D segmentation model and process the data to count the number of nerves segmented.

**Reply by ayafa** [reply]
Is there a way my research partner and I can reach out to you by email to send over an example and explain better? Thank you!

**Reply by hermancollin** [reply]
@ayafa you can reach us at axondeepseg@googlegroups.con
